modules/data-entry/src/main/java/ca/sickkids/ccm/lfs/PaginationServlet.java

```diff
@@ -484,11 +486,14 @@ private String sanitizeComparator(String comparator)
      */
     private void writeSummary(final JsonGenerator jsonGen, final SlingHttpServletRequest request, final long[] limits)
     {
+        final boolean isApprox = (limits[3] > (QUERY_SIZE_MULTIPLIER * limits[1]));
         jsonGen.write("req", request.getParameter("req"));
         jsonGen.write("offset", limits[0]);
         jsonGen.write("limit", limits[1]);
         jsonGen.write("returnedrows", limits[2]);
-        jsonGen.write("totalrows", (limits[3] > limits[1]) ? -1 : (limits[0] + limits[2]));
+        jsonGen.write("totalrows", isApprox
+            ? ((QUERY_SIZE_MULTIPLIER * limits[1]) + limits[0]) : (limits[0] + limits[3]));
+        jsonGen.write("isApprox", isApprox);
```

Comment:
I suggest an even more descriptive field name:
```suggestion
        jsonGen.write("totalIsApproximate", isApprox);
```

Reply:
Implemented by commit `9925415` :heavy_check_mark: